simtools/corsika/corsika_output.py

```diff
@@ -0,0 +1,1301 @@
+        """
+        The configuration of the histograms.
+
+        Returns
+        -------
+        dict:
+            the dictionary with the histogram configuration.
+        """
+        if self._hist_config is None:
+            msg = "No configuration was defined before. The default config is being created now."
+            self._logger.warning(msg)
+            self._hist_config = self._create_histogram_default_config()
+        return self._hist_config
+
+    @hist_config.setter
+    def hist_config(self, input_config):
+        """
+        Set the configuration for the histograms (e.g., bin size, min and max values, etc).
+        The input is allowed either through a yaml file or a dictionary. If nothing is given,
+        the dictionary is created with default values.
+
+        Parameters
+        ----------
+        input_config: str, Path, dict or NoneType
+            yaml file with the configuration parameters to create the histograms. For the correct
+            format, please look at the docstring at `_create_histogram_default_config`.
+            Alternatively, it can be a dictionary with the configuration parameters to create
+            the histograms.
+        """
+        input_dict, input_yaml = None, None
+        if isinstance(input_config, dict):
+            input_dict = input_config
+        else:
+            input_yaml = input_config
+        self._hist_config = collect_data_from_yaml_or_dict(input_yaml, input_dict, allow_empty=True)
+
+    def hist_config_to_yaml(self, file_name=None):
+        """
+        Save the histogram configuration dictionary to a yaml file.
+
+        Parameters
+        ----------
+        file_name: str
+            Name of the output file, in which to save the histogram configuration.
+
+        """
+
+        if file_name is None:
+            file_name = "hist_config"
+        file_name = Path(file_name).with_suffix(".yml")
+        output_config_path = self.io_handler.get_output_directory(self.label, "corsika")
+        output_config_file = output_config_path.joinpath(file_name)
+        save_dict_to_file(self.hist_config, output_config_file)
+
+    def _create_histogram_default_config(self):
+        """
+        Create a dictionary with the configuration necessary to create the histograms. It is used
+        only in case the configuration is not provided in a yaml file or dict.
+
+        Three histograms are created: hist_position with 3 dimensions (x, y positions and the
+        wavelength), hist_direction with 2 dimensions (directive cosinus in x and y directions),
+        hist_time_altitude with 2 dimensions (time and altitude of emission).
+
+        Four arguments are passed to each dimension in the dictionary:
+
+        "bins": the number of bins,
+        "start": the first element of the histogram,
+        "stop": the last element of the histogram, and
+        "scale" to define the scale of the bins which can be "linear" or "log". If "log",
```

Comment:
Is log == log_10 or log == ln? I would prefer to see this written explicitly in the comment.